packages/fast-tooling-react/src/utilities/ajv-validation.ts

```diff
@@ -0,0 +1,83 @@
+import ajv, { ErrorObject, ValidateFunction } from "ajv";
+
+import stringify from "fast-json-stable-stringify";
+
+/**
+ * Dictionary of Ajv validators. One validator per unique schema key.
+ * @type {{ [key: string]: ajv.Ajv }}
+ */
+let ajvDictionary: { [key: string]: ajv.Ajv } = {};
+
+/**
+ * Dictionary of Ajv validate functions. This is compiled schema cache. One per unique schema key.
+ * @type {{ [key: string]: ValidateFunction }}
+ */
+let validateFunctionDictionary: { [key: string]: ValidateFunction } = {};
+
+/**
+ * Validate data against given schema.
+ * @method
+ * @param schema {any} schema for the data to be validate against.
+ * @param data {any} data object for validation.
+ * @returns {boolean | PromiseLike<any>} Boolean if data is valid or promise like.
+ */
+export function validateData(schema: any, data: any): boolean | PromiseLike<any> {
+    const schemaRefKey: string = stringify(schema);
+    let dataValidationResult: boolean | PromiseLike<any>;
+
+    if (validateFunctionDictionary[schemaRefKey]) {
+        dataValidationResult = validateFunctionDictionary[schemaRefKey](data) || false;
+    } else {
+        ajvDictionary[schemaRefKey] = new ajv({ schemaId: "auto", allErrors: true });
+        validateFunctionDictionary[schemaRefKey] = ajvDictionary[schemaRefKey].compile(
+            schema
+        );
+        dataValidationResult = validateFunctionDictionary[schemaRefKey](data) || false;
+    }
+
+    return dataValidationResult;
+}
```

Comment:
nit: no need to store the result, you could just do:

```ts
const schemaRefKey: string = stringify(schema);

if (!validateFunctionDictionary[schemaRefKey]) {
    ajvDictionary[schemaRefKey] = new ajv({ schemaId: "auto", allErrors: true });
    validateFunctionDictionary[schemaRefKey] = ajvDictionary[schemaRefKey].compile(
        schema
    );
}

return validateFunctionDictionary[schemaRefKey](data) || false;
```

Reply:
Nice thanks :)